applications/ShapeOptimizationApplication/custom_utilities/mapping/mapper_vertex_morphing_symmetric.cpp

```diff
@@ -0,0 +1,363 @@
+}
+
+void MapperVertexMorphingSymmetric::Map( const Variable<double> &rOriginVariable, const Variable<double> &rDestinationVariable)
+{
+    KRATOS_ERROR << "Scalar mapping not possible." << std::endl;
+}
+
+void MapperVertexMorphingSymmetric::InverseMap( const Variable<array_3d> &rDestinationVariable, const Variable<array_3d> &rOriginVariable)
+{
+    if (mIsMappingInitialized == false)
+        Initialize();
+
+    BuiltinTimer timer;
+    KRATOS_INFO("") << std::endl;
+    KRATOS_INFO("ShapeOpt") << "Starting inverse mapping of " << rDestinationVariable.Name() << "..." << std::endl;
+
+    Vector values_origin(mrOriginModelPart.Nodes().size()*3);
+    Vector values_destination(mrDestinationModelPart.Nodes().size()*3);
+
+    // Prepare vectors for mapping
+    values_origin.clear();
+    values_destination.clear();
+
+    block_for_each(mrDestinationModelPart.Nodes(), [&](const ModelPart::NodeType& rNode) {
+        const int i = rNode.GetValue(MAPPING_ID);
+        const array_3d& r_nodal_variable = rNode.FastGetSolutionStepValue(rDestinationVariable);
+        values_destination[i*3+0] = r_nodal_variable[0];
+        values_destination[i*3+1] = r_nodal_variable[1];
+        values_destination[i*3+2] = r_nodal_variable[2];
+    });
+
+    SparseSpaceType::TransposeMult(mMappingMatrix,values_destination,values_origin);
+
+    // Assign results to nodal variable
+    block_for_each(mrOriginModelPart.Nodes(), [&](ModelPart::NodeType& rNode) {
+        const int i = rNode.GetValue(MAPPING_ID);
+        array_3d& r_node_vector = rNode.FastGetSolutionStepValue(rOriginVariable);
+        r_node_vector(0) = values_origin[i*3+0];
+        r_node_vector(1) = values_origin[i*3+1];
+        r_node_vector(2) = values_origin[i*3+2];
+    });
+
+    KRATOS_INFO("ShapeOpt") << "Finished mapping in " << timer.ElapsedSeconds() << " s." << std::endl;
+}
+
+void MapperVertexMorphingSymmetric::InverseMap(const Variable<double> &rDestinationVariable, const Variable<double> &rOriginVariable)
+{
+    KRATOS_ERROR << "Scalar mapping not possible." << std::endl;
+}
+
+void MapperVertexMorphingSymmetric::Update()
+{
+    if (mIsMappingInitialized == false) {
+        KRATOS_ERROR << "Mapping has to be initialized before calling the Update-function!";
+    }
+
+    BuiltinTimer timer;
+    KRATOS_INFO("ShapeOpt") << "Starting to update mapper..." << std::endl;
+
+    InitializeMappingVariables();
+    AssignMappingIds();
+
+    if (mMapperSettings["plane_symmetry"].GetBool()) {
+        mpSymmetry = Kratos::make_unique<SymmetryPlane>(mrOriginModelPart, mrDestinationModelPart, mMapperSettings["plane_symmetry_settings"]);
+    } else if (mMapperSettings["revolution"].GetBool()) {
+        mpSymmetry = Kratos::make_unique<SymmetryRevolution>(mrOriginModelPart, mrDestinationModelPart, mMapperSettings["revolution_settings"]);
+    } else {
+        KRATOS_ERROR << "No symmetry type specified" << std::endl;
+    }
+
+    InitializeComputationOfMappingMatrix();
```

Comment:
For me `InitializeComputationOfMappingMatrix` and `CreateSearchTreeWithAllNodesInOriginModelPart` are part of 
`ComputeMappingMatrix` function. Is there a reason why they are called outside of `ComputeMappingMatrix` ?

Reply:
Just because i have reused the structure of the already existing mappers.

Reply:
Indeed it would be nicer if both functions are included in `ComputeMappingMatrix`. But since it affects also the other mappers it would probably be a separate PR.

Reply:
> Indeed it would be nicer if both functions are included in ComputeMappingMatrix

I can move them in there, no prob.